### PR TITLE
docs(examples/nat): add an example which use a nat gateway and vpc peering

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:
Add an example which use a nat gateway and vpc peering to enable communications between vpcs and the internet.
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [x] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
None
```

### CODE REVIEW

File: examples/nat/nat_gateway-vpc_peering/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  > Add a DNAT rule for subnet B, set scenario to Direct Connect/Cloud Connect and enter an IP address of
+  a server in subnet B for private IP address.
+
+* a security group:
+  > a security group with two ingress rules.
+
+* a ECS instance:
+  > VPC A DNAT rule associate a ECS instance and VPC B DNAT rule associate a ECS instance.
+
+a VPC peering:
```
Comment (Lance52259) [inline examples/nat/nat_gateway-vpc_peering/README.md]:
```suggestion
* a VPC peering:
```

File: examples/nat/nat_gateway-vpc_peering/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+VPC A and you can add SNAT and DNAT rules for internet connectivity, subnet B connects to subnet A through a VPC
+peering connection and uses the public NAT gateway of subnet A to communicate with the internet.
+
+* a VPC instance:
+  > Create a VPC A instance with subnet A. Create a VPC B instance with subnet B.
+
+* a NAT gateway:
+  > The NAT gateway is created in VPC A, and subnet B can use the NAT gateway to communicate the internet.
+
+* a SNAT rule:
```
Comment (Lance52259) [inline examples/nat/nat_gateway-vpc_peering/README.md]:
```suggestion
* an SNAT rule:
```

File: examples/nat/nat_gateway-vpc_peering/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  > Add a DNAT rule for subnet A, select VPC for scenario and enter an IP address of a server in subnet A
+  for private IP address.
+  > Add a DNAT rule for subnet B, set scenario to Direct Connect/Cloud Connect and enter an IP address of
+  a server in subnet B for private IP address.
+
+* a security group:
+  > a security group with two ingress rules.
+
+* a ECS instance:
+  > VPC A DNAT rule associate a ECS instance and VPC B DNAT rule associate a ECS instance.
```
Comment (Lance52259) [inline examples/nat/nat_gateway-vpc_peering/README.md]:
```suggestion
* an ECS instance:
  > VPC A DNAT rule associate a ECS instance and VPC B DNAT rule associate an ECS instance.
```

Comment (ruwenqiang123) [reply examples/nat/nat_gateway-vpc_peering/README.md]: done

File: examples/nat/nat_gateway-vpc_peering/README.md (hunk 2 above)
Comment (ruwenqiang123) [reply examples/nat/nat_gateway-vpc_peering/README.md]: done

File: examples/nat/nat_gateway-vpc_peering/README.md (hunk 1 above)
Comment (ruwenqiang123) [reply examples/nat/nat_gateway-vpc_peering/README.md]: done

File: examples/nat/nat_gateway-vpc_peering/main.tf [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+  network {
+    uuid = huaweicloud_vpc_subnet.subnetA.id
+  }
+}
+
+resource "huaweicloud_nat_dnat_rule" "dnatA" {
+  nat_gateway_id        = huaweicloud_nat_gateway.mygateway.id
+  floating_ip_id        = huaweicloud_vpc_eip.eipC.id
+  port_id            = huaweicloud_compute_instance.instanceA.network[0].port
```
Comment (Lance52259) [inline examples/nat/nat_gateway-vpc_peering/main.tf]:
```suggestion
  floating_ip_id        = huaweicloud_vpc_eip.eipC.id
  port_id               = huaweicloud_compute_instance.instanceA.network[0].port
```

File: examples/nat/nat_gateway-vpc_peering/main.tf [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  source_type    = 1
+  cidr           = "192.168.2.0/24"
+}
+
+data "huaweicloud_compute_flavors" "flavorB" {
+  availability_zone = data.huaweicloud_availability_zones.myaz.names[0]
+  performance_type  = "normal"
+  cpu_core_count    = 2
+  memory_size       = 4
+}
```
Comment (Lance52259) [inline examples/nat/nat_gateway-vpc_peering/main.tf]: The flavor A and flavor B are same, why shall we use two data source to query it.

Comment (ruwenqiang123) [reply examples/nat/nat_gateway-vpc_peering/main.tf]: done

File: examples/nat/nat_gateway-vpc_peering/main.tf (hunk 4 above)
Comment (ruwenqiang123) [reply examples/nat/nat_gateway-vpc_peering/main.tf]: done

**Pull request merged**